### PR TITLE
Port over remaining Lavaland ruins

### PR DESCRIPTION
**Pull request by Markolie**
**What does this PR do:**
This pull request ports over the remaining Lavaland ruins from /tg/ and all associated items.

**Ruins:**
**The Wishgranter Cube**: A cube of unsimulated walls with a wish granter inside.
**Ruin of Envy**: A ruin with an envy knife, which upon attacking a human turns you into that human.
**Ruin of Gluttony**: A ruin with walls that can only be moved through when fat. Contains a syringe with Gluttony Toxin, which eventually turns you into a morph.
**Ruin of Greed**: A ruin which spawns a special dice, which has ten varying bad effects and ten varying good effects.
**Ruin of Pride**: A ruin with a mirror which lets you change your appearance, but upon finishing the changes will throw you down a chasm and teleports you to a different z-level.
**Ruin of Sloth**: A ruin full of sepia tiles that take very long to traverse. Upon the end, there's a surprise.
**Dead God**: A ruin with a dead Ratvar and some clockcult items.
**UFO Crash**: A crashed abductor UFO.
**Xenomorph Nest**: A nest containing, unsurprisingly, AI xenomorphs.
**Fountain Hall**: A hall with a fountain which heals you.
**Survival Capsule**: A survival capsule with a dead miner inside.
**Summoning Ritual**: A cult ritual gone wrong.
**Ripley**: A stranded ripley.
**Ancient Puzzle**: An ancient puzzle, which upon solved gives you a cube that can put a restrained player into a puzzle. The puzzle needs to be solved to save that person.

**Changelog:**
:cl: Markolie
add: All remaining Lavaland ruins have been ported over from /tg/.
/:cl:

### CODE REVIEW

**Comment by Kyep** [comment]
The description of this needs to summarize the ruins being ported.
Simply saying "the TG ruins" is not really enough.

**Comment by Markolie** [comment]
Done.

**Comment by Kyep** [comment]
So:
-  Cube: does the wishgranter in this cube let you become a hijack antag? If so, that's not something we want.
- Ruin of Gluttony: does becoming a morph make you an antag? If so.... this is another free ticket to antag.
- Ruin of Greed: You need to list the effects.
- Ruin of Pride: What are the restrictions on the z-levels you can end up on? I assume ending on z2 is not possible?
- Ruin of Sloth: What is the surprise? PR descriptions are not the place to be vague.
- Dead God: Are these clockcult items functional?
- UFO Crash: Does this include abductor tools/weapons? Are those usable by normal players?
- Fountain Hall: What are the limits on using this? Is it a full rejuv? Does it contain a liquid you can bottle and use on yourself later?
- Summoning Ritual: More details please.
- Ripley: does the ripley come with equipment? What equipment?

**Comment by Markolie** [comment]
I'm not a big fan of ruining all the ruins entirely, but here you go: 

- Cube: does the wishgranter in this cube let you become a hijack antag? If so, that's not something we want.
Yes, but the ruin can just be blacklisted through the config.

- Ruin of Gluttony: does becoming a morph make you an antag? If so.... this is another free ticket to antag.
It does. Can also be blacklisted if need be.

- Ruin of Greed: You need to list the effects.
1: User is dusted.
2: User dies.
3: Hostile creatures spawn around the user.
4: All equipment on the user is destroyed.
5: The user is monkeyized.
6: The user is slowed down.
7: The user is stunned, takes damage and thrown away.
8: Fuel tank explosion.
9: User gets a cold.
10: Nothing.
11: A cookie spawns.
12: The user is fully healed.
13: A lot of money spawns.
14: A mateba spawns.
15: A random spellbook spawns.
16: The user receives their own player-controlled butler they can summon.
17: A random traitor kit spawns.
18: A spare captain ID spawns.
19: User takes half brute, burn and cold damage.
20: User becomes a wizard.

- Ruin of Pride: What are the restrictions on the z-levels you can end up on? I assume ending on z2 is not possible?
Any z-level you can teleport to, so the admin level is excluded.

- Ruin of Sloth: What is the surprise? PR descriptions are not the place to be vague.
The surprise is that there's nothing at the end.

- Dead God: Are these clockcult items functional?
Not really, it's just brass stuff.

- UFO Crash: Does this include abductor tools/weapons? Are those usable by normal players?
There's three alien surgery tools they can use.

- Fountain Hall: What are the limits on using this? Is it a full rejuv? Does it contain a liquid you can bottle and use on yourself later?
It's omnizine with a higher overdose threshold. Can not be bottled.

- Summoning Ritual: More details please.
It's just a fluff ruin.

- Ripley: does the ripley come with equipment? What equipment?
No equipment.

**Comment by Kyep** [comment]
Okay. I'll vote for this, though we heads might decide we want the first two ruins blacklisted, as traditionally we don't allow items that make you an antag as loot, especially not items that give you hijack.

**Comment by Markolie** [comment]
Yea, that's fine. They're not particularly interesting ruins anyway.

**Comment by MrMagolor** [comment]
Nat 20s here we come

**Comment by MrMagolor** [comment]
Also, with pride ruin: would becoming an antag race (ie shadowling) make you actually an antag, or just a regular crew with antag powers?

**Comment by Markolie** [comment]
You can't become antag-specific races. Just ones that are regularly available.

**Comment by MrMagolor** [comment]
>You can't become antag-specific races. Just ones that are regularly available.

Does that mean no Vox Armalis, or even becoming a race you don't have karma-unlocked? Kinda pointless then.

**Comment by Markolie** [comment]
You can become karma-locked races. Vox Armalis are very bugged, so that'd be a bad idea.

**Comment by MrMagolor** [comment]
Can I presume that you have to somehow power the abductor ruin to be able to get organs from it? Or does it start with not requiring power?

**Comment by Markolie** [comment]
I don't believe the abductor ruin has the organ machine.

File: _maps/map_files/RandomRuins/LavaRuins/lavaland_surface_ufo_crash.dmm [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+/obj/item/scalpel/alien,
+/obj/item/surgical_drapes,
+/turf/unsimulated/floor/abductor,
+/area/ruin/unpowered)
+"o" = (
+/obj/item/retractor/alien,
+/turf/unsimulated/floor/abductor,
+/area/ruin/unpowered)
+"p" = (
+/obj/machinery/abductor/gland_dispenser,
```
Comment (MrMagolor) [inline _maps/map_files/RandomRuins/LavaRuins/lavaland_surface_ufo_crash.dmm]: Seems like the crashed mothership still has the organ dispenser to me.

Comment (Markolie) [reply _maps/map_files/RandomRuins/LavaRuins/lavaland_surface_ufo_crash.dmm]: Ah. From the looks of it it doesn't use power.

**Comment by MrMagolor** [comment]
On an unrelated note, this PR should probably be reopened for appveyor? I could be mistaken.

**Comment by Bram961** [comment]
For clarity @Markolie @Kyep I calculated the probabilities for antags and whatnot happening due to this PR

https://cdn.discordapp.com/attachments/410782260887945226/610132474638368799/unknown.png

Going off the assumption that
1: There are about 10 ruins per game (I used one shift to count)
2: Ruins can occur multiple times
3: All ruins are found every shift

The chances of certain things occuring are
A wizard: 2%
A traitor: 32%
A morph: 32%
A wizard and or a traitor: 34%
A wizard, traitor, morph or a combination of the three: 56%

**Comment by Markolie** [comment]
Those odds are wildly inaccurate. First of all:

1. Ten ruins is a lot. Bigger ruins cost a lot more than smaller ones and prevent others from spawning. 
2. The odds of retrieving the dice are incredibly low. You have to spend a lot of time gambling which causes brain damage. 
3. The dice has a 1/20 chance of creating a wizard. It had a 9/20 chance of causing bad stuff, 2/20 of which kills you without letting you be cloned. 
4. The gluttony syringe is hard to obtain. Not to mention it just turns you into a morph. 
5. The wishgranter ruin is disabled. 
6. Not all ruins are explored every round. Maybe half of them are. 

This feels a lot like fear mongering based on as I said wildly inaccurate, non-mathematical odds.

**Comment by Bram961** [comment]
You might want to include the full functionality of the dice and the disabling of the wish granter on the PR in that case

The 10 is based on two shifts so its not a big sample but I suppose the chance both shifts deviated majorly from the norm is a small chance as well.

Anyway the point of my post was moreso to show that under absolutely perfect circumstances there is a 32% chance of a round having a traitor due to the wish granter

**Comment by Markolie** [comment]
I like them too, but there’s zero chance of convincing them regardless. I think we both know that 😛.

**Comment by Kyep** [comment]
@Markolie When you say "the wishgranter ruin is disabled" what do you mean? Do you mean its disabled in the code? Or via a config blacklist? Thus far I have NOT updated the ruin blacklist to block any ruins added in this PR.

**Comment by Markolie** [comment]
The latter.

**Pull request merged**